agent/src/test/java/com/teamscale/jacoco/agent/TestBase.java

```diff
@@ -0,0 +1,24 @@
+package com.teamscale.jacoco.agent;
+
+import com.teamscale.jacoco.agent.util.AgentUtils;
+import org.junit.jupiter.api.AfterAll;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Base class that should be used for all tests.
+ * Removes coverage directories created by the tests.
+ */
+public class TestBase {
```

Comment:
a lot of tests don't create coverage directories, so I don't like this solution as it makes it impossible to inherit from some other base class, should we ever need that. We had that in ConQAT times where everything needed to be a `CCSMTestCaseBase` which was cumbersome at times. So I'd like to avoid this if possible.

Could you please try to turn this into a JUnit 5 `@Rule` and use this only in the tests that actually create such directories? (I'm guessing: only integration/system tests).

Alternatively: change the code so the tests write to a temp directory (either the OS temp dir or a `test-tmp` directory that is .gitignore'd)

Reply:
Removed the class and added the `@Afterall` to the two classes causing problems. Probably not the most elegant solution, but I'd like to get the ticket resolved rather quick, so that's my compromise. If you think that we want to do it a different way (with JUnit5 Extensions for example, @Rule doesn't exist anymore) please create a new maintenance ticket for it.